core/dbt/task/debug.py

```diff
@@ -125,6 +125,10 @@ def run(self):
         self.test_dependencies()
         self.test_connection()
 
+        if self.any_failure:
+            print(red('{}/4 checks failed'.format(len(self.messages))))
```

Comment:
Clever approach, minimally invasive, using the `dbt.ui` color functions—I like it!

One easily fixable problem: `_log_project_fail()` and `_log_profile_fail()` do their own `print`ing directly, rather than appending to the `self.messages` queue. They should append to `messages` so that we can count failures appropriately.

One additional thought: Rather than hard-coding `4` here as the total number of checks, we could keep a running tally that increments with each check we perform. I'm thinking about this in two ways:
- If the specified `profile` is invalid, we won't run a connection test, so really the user will only see 3 checks, 1 of which failed
- The proposal in #3735 would enable a more "targeted" debug that runs fewer checks

Reply:
Following both recommendations above, I've pasted below a quick diff to make 'em happen. You don't have to use—or, feel free to use and rename any/all variables—I just wanted to make sure that I was asking for easily doable things.
```
$ git diff core/dbt/task/debug.py
diff --git a/core/dbt/task/debug.py b/core/dbt/task/debug.py
index 0cb1e8960..0e53518ac 100644
--- a/core/dbt/task/debug.py
+++ b/core/dbt/task/debug.py
@@ -91,6 +91,7 @@ class DebugTask(BaseTask):
         self.project_fail_details = ''
         self.any_failure = False
         self.messages: List[str] = []
+        self.check_counter: int = 0

     @property
     def project_profile(self):
@@ -125,13 +126,12 @@ class DebugTask(BaseTask):
         self.test_dependencies()
         self.test_connection()

+        for message in self.messages:
+            print(message)
         if self.any_failure:
-            print(red('{}/4 checks failed'.format(len(self.messages))))
+            print(red('{}/{} checks failed'.format(len(self.messages), self.check_counter)))
         else:
             print(green('All checks passed!'))
-        for message in self.messages:
-            print(message)
-            print('')

         return not self.any_failure

@@ -289,6 +289,7 @@ class DebugTask(BaseTask):
         return green('OK found and valid')

     def test_git(self):
+        self.check_counter += 1
         try:
             dbt.clients.system.run_cmd(os.getcwd(), ['git', '--help'])
         except dbt.exceptions.ExecutableError as exc:
@@ -319,26 +320,34 @@ class DebugTask(BaseTask):
         self._log_profile_fail()

     def _log_project_fail(self):
+        self.check_counter += 1
         if not self.project_fail_details:
             return

         self.any_failure = True
         if self.project_fail_details == FILE_NOT_FOUND:
             return
-        print('Project loading failed for the following reason:')
-        print(self.project_fail_details)
-        print('')
+        msg = (
+            f'Project loading failed for the following reason: ' \
+            f'\n {self.project_fail_details} ' \
+            f'\n'
+        )
+        self.messages.append(msg)

     def _log_profile_fail(self):
+        self.check_counter += 1
         if not self.profile_fail_details:
             return

         self.any_failure = True
         if self.profile_fail_details == FILE_NOT_FOUND:
             return
-        print('Profile loading failed for the following reason:')
-        print(self.profile_fail_details)
-        print('')
+        msg = (
+            f'Profile loading failed for the following reason: ' \
+            f'\n {self.profile_fail_details} ' \
+            f'\n'
+        )
+        self.messages.append(msg)

     @staticmethod
     def attempt_connection(profile):
@@ -359,6 +368,7 @@ class DebugTask(BaseTask):
         return None

     def _connection_result(self):
+        self.check_counter += 1
         result = self.attempt_connection(self.profile)
         if result is not None:
             self.messages.append(result)
```

Reply:
Instead of keeping a running tally, would it be fair to just say `1 check failed` and trade the accurate count problem for a pluralisation problem?

Reply:
Yes... that's just simpler overall, and generally better, isn't it?